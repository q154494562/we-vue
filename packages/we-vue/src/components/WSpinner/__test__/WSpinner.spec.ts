import { shallowMount } from '@vue/test-utils'
import Spinner from '../'

describe('spinner', () => {
  test('create', () => {
    const wrapper = shallowMount(Spinner, {
      propsData: {},
    })

    expect(wrapper.name()).toBe('w-spinner')
    expect(wrapper.html()).toMatchSnapshot()

    // default type
    expect(wrapper.classes()).toContain('weui-loading')

    wrapper.setProps({
      type: 'snake',
    })
    expect(wrapper.classes()).toContain('wv-spinner')
  })

  test('size', () => {
    const wrapper = shallowMount(Spinner, {
      propsData: {},
    })

    // default size should be 17
    expect(wrapper.vm.size).toBe(17)

    wrapper.setProps({
      size: 20,
    })
    expect(wrapper.vm.size).toBe(20)
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('color', () => {
    const wrapper = shallowMount(Spinner, {
      propsData: {},
    })

    // default color should be '#aaa'
    expect(wrapper.vm.color).toBe('#aaa')

    wrapper.setProps({
      color: 'red',
    })
    expect(wrapper.vm.color).toBe('red')
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('icon font class should be computed correctly', () => {
    const wrapper = shallowMount(Spinner, {
      propsData: {},
    })

    // default fontclass should be ''
    expect(wrapper.vm.fontclass).toBe('')

    wrapper.setProps({
      type: 'snake',
    })
    expect(wrapper.vm.fontclass).toBe('icon-spinner-1')

    wrapper.setProps({
      type: 'double-snake',
    })
    expect(wrapper.vm.fontclass).toBe('icon-spinner9')

    wrapper.setProps({
      type: 'bar-circle',
    })
    expect(wrapper.vm.fontclass).toBe('icon-spinner2')

    wrapper.setProps({
      type: 'dot-circle',
    })
    expect(wrapper.vm.fontclass).toBe('icon-spinner1')
  })
})
