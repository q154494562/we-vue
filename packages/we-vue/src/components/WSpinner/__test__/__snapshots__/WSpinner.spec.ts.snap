// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`spinner color 1`] = `<i class="weui-loading" style="font-size: 17px; color: red;"></i>`;

exports[`spinner create 1`] = `<i class="weui-loading" style="font-size: 17px; color: rgb(170, 170, 170);"></i>`;

exports[`spinner size 1`] = `<i class="weui-loading" style="font-size: 20px; color: rgb(170, 170, 170);"></i>`;
